src/utils/firebaseUtils.js

```diff
@@ -1,11 +1,25 @@
 import { db } from '../lib/firebase';
-import { collection, addDoc } from '@firebase/firestore';
+import { setDoc, doc, updateDoc, getDoc } from '@firebase/firestore';
 
-export const addProduct = ({ productName, timeFrame, lastPurchaseDate }) =>
-  addDoc(collection(db, 'products'), {
-    productName,
-    timeFrame,
-    lastPurchaseDate,
-    userToken: Math.floor(Math.random() * 11),
-    date: Date.now(),
+export const createListToken = (listToken) =>
+  setDoc(doc(db, 'listTokens', listToken), {});
+
+export const addProduct = ({
+  productName,
+  timeFrame,
+  lastPurchaseDate,
+  listToken,
+}) =>
+  updateDoc(doc(db, 'listTokens', listToken), {
```

Comment:
I'm a little confused about it because the `addProduct` function is **_updating the data_**. I don't know if you try to create the `updateProduct` function or did you have a problem with `add` versus `update`.

Can you please give us more context about your solution?